src/pineko/evolve.py

```diff
@@ -148,28 +151,60 @@ def evolve_grid(
         renormalization scale variation
     xif : float
         factorization scale variation
-    alphas_values : None or list
-        values of strong coupling used to collapse grids
     assumptions : str
         assumptions on the flavor dimension
     comparison_pdf : None or str
         if given, a comparison table (with / without evolution) will be printed
     """
-    _x_grid, _pids, mur2_grid, _muf2_grid = grid.axes()
+    x_grid, _pids, mur2_grid, _muf2_grid = grid.axes()
+    sv_method = None
+    if operators.operator_card.configs.scvar_method is not None:
+        sv_method = operators.operator_card.configs.scvar_method.name
+    xif = 1.0 if sv_method == "EXPANDED" else xif
+    tcard = operators.theory_card
+    opcard = operators.operator_card
+    # rotate the targetgrid
+    eko.io.manipulate.xgrid_reshape(
+        operators, targetgrid=eko.interpolation.XGrid(x_grid)
+    )
     check.check_grid_and_eko_compatible(grid, operators, xif)
     # rotate to evolution (if doable and necessary)
-    if np.allclose(operators.rotations.pids, br.flavor_basis_pids):
-        eko.output.manipulate.to_evol(operators)
-    elif not np.allclose(operators.rotations.inputpids, br.evol_basis_pids):
+    if np.allclose(operators.rotations.inputpids, br.flavor_basis_pids):
+        eko.io.manipulate.to_evol(operators)
+    # Here we are checking if the EKO contains the rotation matrix (flavor to evol)
+    elif not np.allclose(operators.rotations.inputpids, br.rotate_flavor_to_evolution):
         raise ValueError("The EKO is neither in flavor nor in evolution basis.")
     # do it
     order_mask = pineappl.grid.Order.create_mask(grid.orders(), max_as, max_al)
-    # TODO this is a hack to not break the CLI
-    # the problem is that the EKO output still does not contain the theory/operators card and
-    # so I can't compute alpha_s *here* if xir != 1
-    if np.isclose(xir, 1.0) and alphas_values is None:
-        mur2_grid = np.array(list(operators.Q2grid))
-        alphas_values = [astrong.a_s(q2) for q2 in operators.Q2grid]
+    # This is really the facto scale grid only for scheme A and C
+    muf2_grid = operators.mu2grid
+    # PineAPPL wants alpha_s = 4*pi*a_s
+    # remember that we already accounted for xif in the opcard generation
+    evmod = eko.couplings.couplings_mod_ev(opcard.configs.evolution_method)
+    # Couplings ask for the square of the masses
+    quark_masses = [(x.value) ** 2 for x in tcard.quark_masses]
+    sc = eko.couplings.Couplings(
+        tcard.couplings,
+        tcard.order,
+        evmod,
+        quark_masses,
+        hqm_scheme=tcard.quark_masses_scheme,
+        thresholds_ratios=tuple(tcard.matching.raw.values()),
```

Comment:
no! this *is* the bug fixed in https://github.com/NNPDF/eko/pull/199
```suggestion
        thresholds_ratios=np.power(list(iter(tcard.matching)), 2.0),
```
see also https://github.com/NNPDF/eko/blob/2a1a26d05b6a36454b8d1e1489e44b983a81a96d/src/eko/runner/legacy.py#L81